examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPool_main.cpp

```diff
@@ -91,14 +141,21 @@ struct Arg : public option::Arg
 enum  optionIndex
 {
     UNKNOWN_OPT,
-    HELP
+    HELP,
+    SAMPLES,
+    INTERVAL
 };
 
 const option::Descriptor usage[] = {
     { UNKNOWN_OPT, 0, "", "",                Arg::None,
       "Usage: CustomPayloadPoolExample <publisher|subscriber>\n\nGeneral options:" },
     { HELP,    0, "h", "help",               Arg::None,      "  -h \t--help  \tProduce help message." },
-    {0, 0, 0, 0, 0, 0}
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nPublisher options:"},
+    { SAMPLES, 0, "s", "samples",            Arg::NumericRange<>,
+      "  -s <num>, \t--samples=<num>  \tNumber of samples (0, default, infinite)." },
+    { INTERVAL, 0, "i", "interval",          Arg::NumericRange<>,
+      "  -i <num>, \t--interval=<num>  \tTime between samples in milliseconds (Default: 100)." },
```

Comment:
Having added this CLI, we should remove the logic that allows me to run the example with `publisher <samples> <interval>` (without the `-s/-i`)